django/core/mail/backends/filebased.py

```diff
@@ -9,13 +9,24 @@
 
 
 class EmailBackend(ConsoleEmailBackend):
-    def __init__(self, *args, file_path=None, **kwargs):
+    def __init__(self, *args, file_path=None, provider=None, **kwargs):
         self._fname = None
-        if file_path is not None:
-            self.file_path = file_path
+
+        if provider is not None:
+            # Being initialized from EMAIL_PROVIDERS.
+            options = settings.EMAIL_PROVIDERS[provider].get("OPTIONS", {})
+            file_path = options.get("file_path")
```

Comment:
An EmailBackend **must not** examine settings.EMAIL_PROVIDERS directly. (That is mail.get_connection()'s responsibility.)

```suggestion
        if provider is not None:
            # Being initialized from EMAIL_PROVIDERS.
            pass
            # file_path arg is already set by mail.get_connection(). Do not override it here.
            # (If you want to raise ImproperlyConfigured for missing file_path, do that here.)
```